data_node/upload_manager.go

```diff
@@ -102,8 +106,86 @@ func (um *UploadManager) handleInitialUpload(w http.ResponseWriter, r *http.Requ
 		return
 	}
 
-	w.WriteHeader(http.StatusCreated)
 	w.Header().Set("ID", id)
+	w.Header().Set("Max-Request-Size", strconv.FormatInt(um.maxChunkSize, 10))
+	w.WriteHeader(http.StatusCreated)
+}
+
+// handleAppendUpload is a function responsible for handling the first upload request
+func (um *UploadManager) handleAppendUpload(w http.ResponseWriter, r *http.Request) {
+	r.Body = http.MaxBytesReader(w, r.Body, um.maxChunkSize)
+	log.Println(um.logPrefix, r.RemoteAddr, "Received APPEND request")
+
+	expectedHeaders := []string{"Content-Length", "Offset", "ID"}
+	err := um.validateUploadHeaders(&r.Header, expectedHeaders...)
+
```

Comment:
Can we please remove the newline? Usually the err check is done right after the statement to avoid confusion. :heavy_check_mark: